js/utils/currency.js

```diff
@@ -26,205 +29,354 @@ export function getEvents() {
+
+  const curMeta = getCurMeta(currency);
+
+  if (curMeta.isFiat) {
+    return 2;
+  } else if (curMeta.isWalletCur) {
+    return curMeta.curData.coinDivisibility;
+  } else if (curMeta.isCryptoListingCur) {
+    return defaultCryptoCoinDivisibility;
+  }
+
+  throw new UnrecognizedCurrencyError();
+}
+
+/*
+ * Based on the provided coin divisibility, will return the minimum value
+ * that coin divisibility supports (e.g. for 8, 1e-8 will be returned).
+ * @param {number} coinDivisibility
+ * @param {object} options
+ * @param {boolean} [options.returnInStandardNotation = false] - if true wil return
+ *   the result in standard notation ('0.00000001' instead of 1e-8). Note the result
+ *   will be returned as a string when this option is true.
+ * @returns {number} - The minimum supported value for the given
+ *   coin divisibility.
+ */
+export function minValueByCoinDiv(coinDivisibility) {
+  const [isValidCoinDiv] = isValidCoinDivisibility(coinDivisibility);
+
+  if (!isValidCoinDiv) {
+    throw new Error('The provided coinDivisibility is not valid.');
+  }
+
+  return 1 / Math.pow(10, coinDivisibility);
 }
 
 /**
- * Converts the amount from an integer to a decimal, rounding to 2 decimal places. If the
- * currency code is for a crypto currency, it will convert from its base units.
+ * Converts the amount from a decimal to an integer based on the provided
+ * coin divisibility.
+ * @param {number|string|BigNumber} value - The number that should be converted to an
+ *   integer.
+ * @param {number} divisibility - An integer representing the coin divisibility (e.g. for
+ *   bitcoin, it is 8)
+ * @returns {BigNumber} - A BigNumber instance representing the integer number.
  */
-export function integerToDecimal(amount, currency, options = {}) {
+export function decimalToInteger(value, divisibility) {
+  validateNumberType(value);
+
+  const [isValidDivis, divisErr] = isValidCoinDivisibility(divisibility);
+
+  if (!isValidDivis) {
+    throw new Error(divisErr);
+  }
+
+  return bigNumber(value)
+    .multipliedBy(
+      bigNumber(10)
+        .pow(divisibility)
+    )
+    .decimalPlaces(0);
+}
+
+/**
+ * Converts the amount from an integer to a decimal based on the provided
+ * divisibility.
+ * @param {number|string|BigNumber} value - The number that should be converted to
+ *   an integer.
+ * @param {number} divisibility - An integer representing the coin divisibility
+ *   (e.g. for bitcoin, it is 8)
+ * @param {object} options
+ * @param {boolean} [options.returnNaNOnError = true] - if true and there's
+ *   an error, rather than an exception being thrown, a BigNumber instance evaluating
+ *   to NaN will be returned. This will allow templates to just display NaN instead of
+ *   bombing on render. It will also allow BigNumber ops (e.g. minus, times, etc...) to
+ *   not bomb.
+ * @returns {BigNumber} - A BigNumber instance representing the decimal number.
+ */
+export function integerToDecimal(value, divisibility, options = {}) {
```

Comment:
So the main change regarding the handling of string based numbers in models is that previously we'd convert a decimal number to and integer on parse and then convert back on sync. Now, the conversion is from a string-based integer to a BigNumber instance.

So when models have a number that represents a currency, they will be handled as BigNumber instances, which are smart enough that if you output them (e.g. `<%= ob.bigNumValue %>` in a template), they will display as a string.

If you need to do arithmetic operations with them, you would use the API that each instance implements (https://github.com/MikeMcl/bignumber.js/) (e.g. bigNumVal.plus(otherBigNumVal)).